libraries/protocol/include/steem/protocol/asset_symbol.hpp

```diff
@@ -63,6 +70,7 @@ class asset_symbol_type
       };
 
       asset_symbol_type() {}
+      operator uint32_t() { return to_nai(); }
```

Comment:
I am a little worried about this introducing subtle bugs later on. uint32_t is typedeffed to a lot of things. What if this was made an explicit operator?

Reply:
I shared the same sentiment, which is why I brought this up to you earlier. However adding explicit to the operator prevents compilation.

Reply:
It doesn't build because there is an implicit conversion (precisely what we want to avoid).

Make the following change:

```c++
/**Comparison operators that allow to return the same object representation
 * for both liquid and vesting symbol/nai.
 */
struct vesting_liquid_less
{
   bool operator ()( const asset_symbol_type& lhs, const asset_symbol_type& rhs ) const
   {
      // Compare as if both symbols represented liquid version.
      return ( lhs.is_vesting() ? lhs.get_paired_symbol() : lhs ) <
             ( rhs.is_vesting() ? rhs.get_paired_symbol() : rhs );
   }

   bool operator ()( const uint32_t& lhs, const asset_symbol_type& rhs ) const
   {
      // Use the other operator, adding the same precision to both NAIs.
      return operator()( asset_symbol_type::from_nai( lhs, rhs.decimals() ), rhs );
   }

   bool operator ()( const asset_symbol_type& lhs, const uint32_t& rhs ) const
   {
      return operator()( lhs, asset_symbol_type::from_nai( rhs, lhs.decimals() ) );
   }
};
```

By removing the double uint32_t comparison we remove the implicit conversion to uint32_t only to convert back to an asset symbol.

Reply:
Makes sense, applying/testing now.